src/Core/Grid/Column/ColumnCollection.php

```diff
@@ -107,14 +107,39 @@ public function toArray()
         return $columns;
     }
 
+    /**
+     * Swap two existing columns.
+     *
+     * @param string $keyFrom the Column ID original position in the Collection
+     * @param string $keyTo the Column ID destination position in the Collection
+     *
+     * @return self
+     */
+    public function swap($keyFrom, $keyTo)
```

Comment:
So you don't change the interface? Isn't it a BC break to add a public method in this class?

Reply:
why?

Reply:
I'd prefer not to use `key`, but `id` instead to keep consistency between other methods.

Reply:
Fine for the "BC break"
Indeed using id would be more consistent with all the other functions parameters

Reply:
You even use "id" in your exception message

Reply:
yup, fair enough: let me update